bobtimus/src/lib.rs

```diff
@@ -395,6 +420,141 @@ mod tests {
+
+            (
+                Client::new(blockchain.node_url.clone().into_string()).unwrap(),
+                blockchain,
+            )
+        };
+        let mining_address = client.getnewaddress().await.unwrap();
+
+        let have_asset_id_alice = client.issueasset(100_000.0, 0.0, true).await.unwrap().asset;
+        let have_asset_id_bob = client.get_bitcoin_asset_id().await.unwrap();
+
+        let mut rate_service = fixed_rate::Service::new();
+        let sell_amount_bob = LiquidBtc::from(Amount::ONE_BTC);
+        let redeem_amount_bob = rate_service
+            .latest_rate()
+            .sell_quote(sell_amount_bob)
+            .unwrap();
+
+        let (
+            fund_address_alice,
+            fund_sk_alice,
+            _fund_pk_alice,
+            fund_blinding_sk_alice,
+            _fund_blinding_pk_alice,
+        ) = make_confidential_address();
+
+        let fund_alice_txid = client
+            .send_asset_to_address(
+                &fund_address_alice,
+                Amount::from(redeem_amount_bob) + Amount::from_btc(10_000.0).unwrap(),
+                Some(have_asset_id_alice),
+            )
+            .await
+            .unwrap();
+        client.generatetoaddress(1, &mining_address).await.unwrap();
+
+        let input_alice = extract_input(
+            &client.get_raw_transaction(fund_alice_txid).await.unwrap(),
+            fund_address_alice,
+        )
+        .unwrap();
+
+        let (
+            final_address_alice,
+            _final_sk_alice,
+            _final_pk_alice,
+            _final_blinding_sk_alice,
+            _final_blinding_pk_alice,
+        ) = make_confidential_address();
+
+        // let address = client.getnewaddress().await.unwrap();
+        // let _txid = client
+        //     .send_asset_to_address(
+        //         &address,
+        //         Amount::from_btc(10.0).unwrap(),
+        //         Some(have_asset_id_bob),
+        //     )
+        //     .await
+        //     .unwrap();
+        // client.generatetoaddress(1, &mining_address).await.unwrap();
+
+        let mut bob = Bobtimus {
+            rng: &mut thread_rng(),
+            rate_service,
+            secp: Secp256k1::new(),
+            elementsd: client.clone(),
+            btc_asset_id: have_asset_id_bob,
+            usdt_asset_id: have_asset_id_alice,
+        };
+
+        let transaction = bob
+            .handle_create_buy_swap(CreateSwapPayload {
+                alice_inputs: vec![AliceInput {
+                    outpoint: input_alice.0,
+                    blinding_key: fund_blinding_sk_alice,
+                }],
+                address: final_address_alice,
+                btc_amount: sell_amount_bob,
+            })
+            .await
+            .unwrap();
+
+        let transaction = swap::alice_finalize_transaction(transaction, {
+            let commitment = input_alice.1.into_confidential().unwrap().value;
+            move |mut tx| async move {
```

Comment:
Does this have to be async?

Reply:
Yep, the second argument (`S`) to this function has to fulfil
```
S: FnOnce(Transaction) -> F,
F: Future<Output = Result<Transaction>>,
```
so this closure must return a future.